### DISCUSSION TITLE
LAG , IGNORING NULLS throws parse error

### DISCUSSION THREAD

**Discussion by rvishureddy** [duckdb/duckdb#11569] [Q&A] [answered]
CREATE TABLE data (
      timestamp TIMESTAMP,
      value INT
  );
  
  INSERT INTO data (timestamp, value) VALUES
      ('2022-01-01 00:00:00', 10),
      ('2022-01-01 00:15:00', NULL),
      ('2022-01-01 00:30:00', NULL),
      ('2022-01-01 00:45:00', 20),
      ('2022-01-01 01:00:00', NULL);
      
      select timestamp, COALESCE(value, LAG(value) OVER (ORDER BY timestamp)) as filled_value from data;
┌─────────────────────┬──────────────┐
 │      timestamp      │ filled_value │
 │      timestamp      │    int32     │
├─────────────────────┼──────────────┤
│ 2022-01-01 00:00:00 │           10 │
│ 2022-01-01 00:15:00 │           10 │
│ 2022-01-01 00:30:00 │                │
│ 2022-01-01 00:45:00 │           20 │
│ 2022-01-01 01:00:00 │           20 │
└─────────────────────┴──────────────┘

And the third row still has null value, 

I tried ignoring nulls but the i see the parser error. Is ignoring nulls not defined in the duckdb?

select timestamp, COALESCE(value, LAG(value) IGNORE NULLS OVER (ORDER BY timestamp)) as filled_value from data;
Error: Parser Error: syntax error at or near "IGNORE"
LINE 1: ... timestamp, COALESCE(value, LAG(value) IGNORE NULLS OVER (ORDER BY timestamp))...

**Comment by soerenwolfers** [comment, marked as answer]
It's ```LAG(value IGNORE NULLS) OVER ...``` as explained at https://duckdb.org/docs/sql/window_functions.html#lagexpr-offset-default-ignore-nulls

**Reply by rvishureddy** [reply]
Thank you that works,
